src/main/java/com/marcusslover/plus/lib/sound/Music.java

```diff
@@ -0,0 +1,213 @@
+				target.stopSound(this.loop.sound);
+				if (this.tail != null) {
+					this.tail.send(target);
+				}
+			}, this.loopLength - (this.ticks % this.loopLength) - 1);
+		} else {
+			target.stopSound(this.loop.sound);
+			if (this.tail != null) {
+				this.tail.send(target);
+			}
+		}
+	}
+
+	/**
+	 * Stop the music from the audience without playing the tail.
+	 *
+	 * @param audience The audience.
+	 */
+	public void stopAll(Audience audience) {
+		this.tickTask.cancel();
+		this.loopTask.cancel();
+		if (this.intro != null)
+			audience.stopSound(this.intro().sound);
+		if (this.loop != null)
+			audience.stopSound(this.loop().sound);
+		if (this.tail != null)
+			audience.stopSound(this.tail().sound);
+	}
+
+	/**
+	 * Stop the music from the command sender without playing the tail.
+	 *
+	 * @param target The command sender.
+	 */
+	public void stopAll(CommandSender target) {
+		this.tickTask.cancel();
+		this.loopTask.cancel();
+		if (this.intro != null)
+			target.stopSound(this.intro().sound);
+		if (this.loop != null)
+			target.stopSound(this.loop().sound);
+		if (this.tail != null)
+			target.stopSound(this.tail().sound);
+	}
+
+	@Override
+	public @NotNull <T extends CommandSender> Music send(@NotNull T target) {
+		if (this.introLength != 0 && this.intro != null) {
+			this.intro.send(target);
+			loopTask = Task.syncRepeating(ServerUtils.getCallingPlugin(), () -> {
+				this.loop.send(target);
+				this.loops++;
+			}, this.introLength + 20, loopLength);
+		} else {
+			loopTask = Task.syncRepeating(ServerUtils.getCallingPlugin(), () -> {
+				this.loop.send(target);
+				this.loops++;
+			}, 0, loopLength);
+		}
+		return this;
+	}
+
+	@Override
+	public @NotNull Music send(Audience audience) {
+		if (this.introLength != 0 && this.intro != null) {
+			this.intro.send(audience);
+			loopTask = Task.syncRepeating(ServerUtils.getCallingPlugin(), () -> {
+				this.loop.send(audience);
+				this.loops++;
+			}, this.introLength + 20, loopLength);
+		} else {
+			loopTask = Task.syncRepeating(ServerUtils.getCallingPlugin(), () -> {
+				this.loop.send(audience);
+				this.loops++;
+			}, 0, loopLength);
+		}
+		return this;
+	}
+
+	/**
+	 * Creates a new music object.
+	 *
+	 * @param note The note to play on loop.
+	 * @param noteLength The length of the note in ticks.
+	 * @return The music object.
+	 */
+	public static @NotNull Music of(@NotNull Note note, long noteLength) {
+		return new Music(null, 0, note, noteLength, null);
+	}
+
+	/**
+	 * Create a new music object with an intro and tail note.
+	 *
+	 * @param intro Note to play before the loop.
+	 * @param introLength Length of the intro in ticks.
+	 * @param loop Note to play on loop.
+	 * @param loopLength Length of the loop in ticks.
+	 * @param tail Note to play when stopped.
+	 * @return The music object.
+	 */
+	public static @NotNull Music of (@NotNull Note intro, long introLength, @NotNull Note loop, long loopLength, @NotNull Note tail) {
```

Comment:
Fails to follow code style of this repo.

Style: https://google.github.io/styleguide/javaguide.html